### ISSUE TITLE
Problem MTEX on new iMac with Mac OS 10.11 El Capitan - solution that works

### ISSUE THREAD

**Issue by DavidMainprice** [mtex-toolbox/mtex#120]
The new “System Integrity Protection” (SIP) of Mac OS 10.11 El Capitan has been disabled and this has been checked twice via the terminal window using "csrutil status".

The error message is as follows ...

MTEX binary check failed!

 The original error message was:
 Error using call_extern (line 158)
Error running external program:

/MatLab Programs/mtex-4.1.4/c/bin/maci64/odf2pf /private/tmp/odf2pf591.txt 2>> /private/tmp/output_ManuelsiMaclocal_MThieme.log

---

We have now found a solution to this problem. Put your mtex-4.1.4 folder in MATLAB folder
You should find this at
Users/ User_Name/ Documents / MATLAB/ mtex-4.1.4

all the best David

**Comment by ralfHielscher** [comment]
Hi David,

is it with this configuration possible to run mtex with the System Integrity Protection enabled?

Ralf.

**Comment by magnusho** [comment]
Hi,

I have the above-mentioned problem when I try to run mtex 4.2.1 on El Capitan. The mtex-4.2.1 folder is located in Users/my_user/Documents/MATLAB/. I've disabled SIP, and the csrutil status returns

System Integrity Protection status: enabled (Custom Configuration).

Configuration:
	Apple Internal: disabled
	Kext Signing: disabled
	Filesystem Protections: disabled
	Debugging Restrictions: disabled
	DTrace Restrictions: disabled
	NVRAM Protections: disabled

Still I get the "binary check failed" message with reference to

/Users/magnush/Documents/MATLAB/mtex-4.2.1/c/bin/maci64/odf2pf

Is there anything I'm missing? 

Thanks!

Magnus

**Comment by DavidMainprice** [comment]
Dear Magnus,

We have some times found it necessary to put MTEX folder in the MATLAB folder on iMacs.

The SIP has to disable of course, attached find a couple of PDFs covering both points.

all the best David

**Comment by magnusho** [comment]
Dear David, 

This is actually my problem. I have disabled SIP and put the MTEX folder in the MATLAB folder (Users/my_user/Documents/MATLAB/) according to the instructions I found, but I still get the "binary check failed" issue.

Is there any other known issue that I could have missed?

Thanks again for your time!

//Magnus

**Comment by DavidMainprice** [comment]
I would try to make a new download of MTEX from the GitHub site

all the best David

**Comment by magnusho** [comment]
Already on the 3rd version since yesterday...
Regards,
Magnus

**Comment by DavidMainprice** [comment]
What is the exact error message you get ?

all the best David

**Comment by magnusho** [comment]
This is what I get on startup:

               MTEX binary check failed!
 
               The original error message was:
               Error using call_extern (line 158)
               Error running external program:

               /Users/magnush/Documents/MATLAB/mtex-4.2.1/c/bin/maci64/odf2pf   
               /private/tmp/odf2pf1601.txt 2>> /private/tmp/output_MagnussMacBookAirlocal_magnush.log

Many functions till works. I can import EBSD data and plot scatterplots etc, trying to calculate the odf returns the same message followed by detailed error messages for functions 

>> odf = calcODF(ebsd.orientations)

              Error using call_extern (line 158)
              Error running external program:

               /Users/magnush/Documents/MATLAB/mtex-4.2.1/c/bin/maci64/odf2fc
               /private/tmp/odf2fc6278.txt 2>>
               /private/tmp/output_MagnussMacBookAirlocal_magnush.log

              Error in unimodalComponent/calcFourier>gcA2fourier (line 67)
              f = call_extern('odf2fc','EXTERN',g,c,A);

              Error in unimodalComponent/calcFourier (line 29)
              f_hat = gcA2fourier(abg,c,A);

              Error in ODF/calcFourier (line 32)
              hat = calcFourier(odf.components{i},L,varargin{:});

              Error in ODF/FourierODF (line 8)
              f_hat = calcFourier(odf,varargin{:});

              Error in orientation/calcFourierODF (line 54)
              odf = FourierODF(odf,get_option(varargin,{'L','bandwidth','fourier'},L,'double'));

              Error in orientation/calcODF (line 64)
              odf = calcFourierODF(ori,varargin{:},'kernel',psi);

**Comment by DavidMainprice** [comment]
Now if I compare this with initial problem with our iMac

initialize MTEX 4.1.4 ....
 MTEX binary check failed!

 The original error message was:
 Error using call_extern (line 158)
Error running external program:

/MatLab Programs/mtex-4.1.4/c/bin/maci64/odf2pf /private/tmp/odf2pf591.txt 2>> /private/tmp/output_ManuelsiMaclocal_MThieme.log

 Check the following options:
 * Compile the binaries yourself.
 * Ask the maintainer for help!
********************************************************************************

This looks very similar to your current message

I suspect if you have moved your MTEX folder MATLAB folder that your SIP is not disabled. When SIP is not disable you get an error associated with access to private folder (> /private/tmp/)

You can find the procedure and other things that need to be updated at:

https://www.dropbox.com/sh/n0lwhkxs4z7tgft/AAC9kf0kerY89-zJWBehjs4sa?dl=0

all the best David

**Comment by magnusho** [comment]
Hi,

As mentioned in my first post, SIP is disabled per the instructions (I included the confirmation from 'csrutil status' in the post) and I have placed the mtex folder in the MATLAB folder. I tried downloading and re-installing mtex several times.  Still the same message unfortunately...

**Comment by zmichels** [comment]
Hi,

I can't quite tell from quickly perusing the earlier posts on this thread... But I had a similar problem when I was installing from folders I downloaded through the repository. It solved everything when I downloaded from the downloads page here:

http://mtex-toolbox.github.io/download

You have probably already done this. However, I just saw this thread and only have a a minute to reply, so I thought I'd throw it out there. I apologize if this is redundant from earlier advice.

Zach

On Feb 12, 2016, at 7:23 AM, magnusho <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi,

As mentioned in my first post, SIP is disabled per the instructions (I included the confirmation from 'csrutil status' in the post) and I have placed the mtex folder in the MATLAB folder. I tried downloading and re-installing mtex several times. Still the same message unfortunately...

-
